traced/ykrt/src/mt.rs

```diff
@@ -139,6 +142,37 @@ impl MT {
             }
         })
     }
+
+    /// Queue `job` to be run on a worker thread.
+    fn queue_job(&self, job: Box<dyn FnOnce() + Send>) {
+        // We have a very simple model of worker threads. Each time a job is queued, we spin up a
+        // new worker thread iff we aren't already running the maximum number of worker threads.
+        // Once started, a worker thread never dies, and spins endlessly waiting for work.
+
+        let inner = Arc::clone(&self.inner);
+        inner.job_queue.lock().unwrap().push_back(job);
+
+        let max_jobs = inner.max_worker_threads.load(Ordering::Relaxed);
+        if inner.active_worker_threads.load(Ordering::Relaxed) < max_jobs {
+            // At the point of the `load` on the previous line, we weren't running the maximum
+            // number of worker threads. There is now a possible race condition where multiple
+            // threads calling `queue_job` could try creating multiple worker threads and push us
+            // over the maximum worker thread limit.
+            if inner.active_worker_threads.fetch_add(1, Ordering::Relaxed) > max_jobs {
```

Comment:
Why don't we only do this check and not the one at line 156 as well? I suppose you are trying to minimise the chance of having to do the correctional subtract?

Reply:
In the average case, we do one load and a compare, which is quicker than the load/sub/add of `fetch_add`.